src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -66,54 +66,53 @@ final class CachedAppConfiguration {
 	private var promises = [(Result<CachedAppConfiguration.AppConfigResponse, Never>) -> Void]()
 	private var requestIsRunning: Bool { !promises.isEmpty }
 
+	private static let queue = DispatchQueue(label: "fetchAppConfiguration queue", qos: .userInitiated, attributes: .concurrent)
+
 	private func getAppConfig(with etag: String? = nil) -> Future<AppConfigResponse, Never> {
 		return Future { promise in
+			Self.queue.async(flags: .barrier) {
+				guard !self.requestIsRunning else {
+					Log.debug("Return immediately because request allready running.", log: .appConfig)
+					Log.debug("Append promise.", log: .appConfig)
+					self.promises.append(promise)
+					return
+				}
 
-			guard !self.requestIsRunning else {
-				Log.debug("Return immediately because request allready running.", log: .appConfig)
 				Log.debug("Append promise.", log: .appConfig)
 				self.promises.append(promise)
-				return
-			}
 
-			Log.debug("Append promise.", log: .appConfig)
-			self.promises.append(promise)
+				func defaultFailureHandler() {
```

Comment:
Can we move the funcs defaultFailureHandler and resolvePromises outside of the getAppConfig func? getAppConfig is hard to read.

Reply:
You wrote `resolvePromises` but sure, I can clean this up.